language/rescue_spec.rb

```diff
@@ -371,11 +371,23 @@ def rescuer.===(exception)
   end
 
   it "evaluates rescue expressions only when needed" do
-    invalid_rescuer = Object.new
     begin
-      :foo
-    rescue invalid_rescuer
-    end.should == :foo
+      ScratchPad << :foo
+    rescue -> { ScratchPad << :bar }.call
```

Comment:
Maybe the lambda should return `StandardError`, so this would be a valid rescue expression in case an exception is raised.